src/python/pants/backend/python/providers/pyenv/rules.py

```diff
@@ -111,6 +121,8 @@ async def get_pyenv_install_info(
         Get(EnvironmentVars, EnvironmentVarsRequest(["PATH"])),
         Get(DownloadedExternalTool, ExternalToolRequest, pyenv_subsystem.get_request(platform)),
     )
+    frozen_env_vars = FrozenDict(pyenv_subsystem.installation_env_vars)
+    installation_fingerprint = hash(frozen_env_vars)
```

Comment:
Hm... I don't think `hash` is stable across Python processes... maybe `CreateDigest` using a string containing the env vars? Or use `hashlib`?

Reply:
Oh hehe that makes a lot of sense 🙃